Wavefront.SDK.CSharp/Common/ReconnectingSocket.cs

```diff
@@ -85,52 +133,90 @@ private void ResetSocket()
         /// </summary>
         /// <param name="message">The message to be sent to the Wavefront proxy.</param>
         public void Write(string message)
-        {
+        {
+            _ = WriteAsync(message);
+        }
+
+        private async Task WriteAsync(string message)
```

Comment:
Is there synchronization between ConnectAsync and WriteAsync?

Reply:
Currently no, which means that writes get dropped when they are concurrent with connect or reset.  I tried letting WriteAsync wait for ResetSocket to finish, but that ends up blocking when there are numerous writes in succession and an unresponsive host.